NitroxClient/GameLogic/Spawning/Metadata/Extractor/FlareMetadataExtractor.cs

```diff
@@ -0,0 +1,23 @@
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+
+namespace NitroxClient.GameLogic.Spawning.Metadata.Extractor;
+
+public class FlareMetadataExtractor : GenericEntityMetadataExtractor<Flare, FlareMetadata>
+{
+    private readonly Items items;
+
+    public FlareMetadataExtractor(Items items)
+    {
+        this.items = items;
+    }
+
+    public override FlareMetadata Extract(Flare flare)
+    {
+        // If the flare is being picked up, its metadata must be set accordingly
+        if (flare.flareActiveState && items.PickingUpObject != flare.gameObject)
```

Comment:
It's very hacky to depend on `PickingUpObject` here; this logic is very easy to break due to its "call-stack" dependency to be correct.

Would it be possible to always pass flare.flareActivateTime to the metadata and handle it in processing of said metadata? I.e. `if value == 0 do nothing`. 

If not, maybe we should add a patch to the flare so that its logic is easier to deal with (and doesn't force us to use spaghetti code).

Reply:
As discussed in the Discord, no other solution was found possible because:
- throwing items is detected at postfix(= flare already thrown)
- picking up items is detected at prefix(= flare still thrown)
and there's no way to differentiate those